Universal FE Randomizer/src/fedata/gba/fe7/FE7Data.java

```diff
@@ -446,6 +446,8 @@ public static CharacterClass valueOf(int classId) {
 				BERSERKER, ASSASSIN, LORD_LYN, BLADE_LORD, ARCHER_F, CLERIC, MAGE_F, TROUBADOUR, PEGASUSKNIGHT, DANCER, SWORDMASTER_F, SNIPER_F,
 				BISHOP_F, SAGE_F, PALADIN_F, VALKYRIE, FALCONKNIGHT, WYVERNLORD_F, LORD_KNIGHT, GREAT_LORD, UBER_SAGE));
 		
+		public static Set<CharacterClass> allPlayerOnlyClasses = new HashSet<CharacterClass>(Arrays.asList(DANCER, BARD));
```

Comment:
I don't actually remember any point in battle when Nils or Ninian are AI controlled, but maybe I'm just not remembering all of FE7.

Reply:
Ninian briefly moves as a green unit during the on-map cutscene in the moments before she joins the party. I'm not familiar with the ROM; this could be scripted rather than true AI control.

https://youtu.be/r8gW18oGOoE?t=787

Reply:
That is correct. If she had moved during an Other Phase, that would be organic AI control. As it stands, that movement is scripted and is part of the pre-battle script.